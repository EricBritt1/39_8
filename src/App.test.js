import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import App from './App';

it('renders without crashing', function() {
    render(<App/>)
});

//This is without any ToDos added on the page. Will fail whenever events are triggered. Will be good to see if actual changes are occuring!
it('matches snapshot', function() {
    const {asFragment} = render(<App />);
    expect(asFragment()).toMatchSnapshot();
});


// I tried my best and even used some tools chatGPT gave me. 
  // I want to be able to update the value of my text input. 
        // The todo SHOULD be added to the page but, react isn't processing the text input due to state i'm sure. This wasn't gone over in the testing tutorial BUT the tutorial did help me make sense of SOME of the problems occuring not all!
it('handles adding a todo', async function() {
  const { asFragment, getByPlaceholderText, getByText } = render(<App />);
  
  const input = getByPlaceholderText("Enter task");
  const button = getByText('Add Todo');
  
  // Set the input value
  input.value = "Wash dishes";

  // Use act to ensure state updates are processed
  act(() => {
    // Click the "Add Todo" button
    fireEvent.click(button);
  });

  // Make your assertions
  expect(asFragment()).toMatchSnapshot();
  expect(button).toBeInTheDocument();
  expect(input).toBeInTheDocument();
});
