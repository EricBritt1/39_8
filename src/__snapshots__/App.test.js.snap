// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles adding a todo 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="todo"
        >
          Enter Todo: 
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="Enter task"
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
      <div>
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="todo"
        >
          Enter Todo: 
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="Enter task"
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
